sdk/bpf/c/inc/solana_sdk.h

```diff
@@ -256,6 +256,11 @@ typedef struct {
   const SolPubkey *program_id; /** program_id of the currently executing program */
 } SolParameters;
 
+/**
+ * Allowance for account data reallocs
+ */
+#define MAX_REALLOC_SIZE (1024 * 10)
```

Comment:
Why is this being called realloc?  It's a new allocation, not a re-allocation.

Reply:
Also why not use `MAX_PERMITTED_DATA_LENGTH`?

Reply:
Because it will also be used to realloc accounts, it's really the incremental amount that an account is allowed to grow via cross-program invocation.  Realloc is limited to just `CreateAccount` for now and only up to `MAX_REALLOC_SIZE`.  More work to do to support generalized realloc, generalized realloc will be supported up to `MAX_PERMITTED_DATA_LENGTH' by increments of `MAX_REALLOC_SIZE`.  Agree, its a little weird and not ideal, but due to the serialized program inputs.

Reply:
>  it's really the incremental amount that an account is allowed to grow via cross-program invocation.

Ah, I see.   Some more comments to that affect would have helped me during my first read.  

What's the serialized program inputs issue?

Reply:
Sounds good, more comments.  All the program inputs are serialized back to back into the program including the account data